### PR TITLE
migration to idpy

### PR DESCRIPTION
**Pull request by peppelinux**
Closes https://github.com/UniversitaDellaCalabria/SATOSA-oidcop/issues/13

satosa-oidcop abandons oidc-op and moves definitively to idpy-oidc

### CODE REVIEW

**Review by melanger — APPROVED** [review]

**Comment by xpavlic** [comment]
Following exception occurs, when I try to run tests.

```ImportError while importing test module '/home/jan/Documents/prace/SATOSA-oidcop/tests/test_oidcop.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_oidcop.py:25: in <module>
    from tests.users import USERS
tests/users.py:5: in <module>
    from satosa.frontends.openid_connect import combine_claim_values
venv/lib/python3.10/site-packages/SATOSA-8.2.0-py3.10.egg/satosa/frontends/openid_connect.py:12: in <module>
    from oic.oic import scope2claims
venv/lib/python3.10/site-packages/oic-1.6.0-py3.10.egg/oic/oic/__init__.py:26: in <module>
    from oic import oauth2
venv/lib/python3.10/site-packages/oic-1.6.0-py3.10.egg/oic/oauth2/__init__.py:20: in <module>
    from oic.oauth2.base import PBase
venv/lib/python3.10/site-packages/oic-1.6.0-py3.10.egg/oic/oauth2/base.py:16: in <module>
    from oic.utils.settings import PyoidcSettings
venv/lib/python3.10/site-packages/oic-1.6.0-py3.10.egg/oic/utils/settings.py:20: in <module>
    from pydantic import BaseSettings
venv/lib/python3.10/site-packages/pydantic-2.0b2-py3.10.egg/pydantic/_migration.py:279: in wrapper
    raise PydanticImportError(f'`{import_path}` has been removed in V2.')
E   pydantic.errors.PydanticImportError: `pydantic:BaseSettings` has been removed in V2.
E   
E   For further information visit https://errors.pydantic.dev/2.0b2/u/import-error
```
It is caused by `pydantic` library when importing `USERS` and `OIDC_USERS` objects from `tests/users.py` in `test_oidcop.py`. 
`pydantic` library is dependecy from SATOSA and from version 2.0 does not support importing of objects. `tests/users.py` is somehow dependent on it through `from satosa.frontends.openid_connect import combine_claim_values`.

I think, that it can be solved by moving `USERS` and `OIDC_USERS` objects to `tests/test_oidcop.py` to get rid of importing them.

**Review by vladimir-mencl-eresearch — COMMENTED** [review]
Sorry, I do not know the codebase enough to review a major change like this....

**Comment by peppelinux** [comment]
> Following exception occurs, when I try to run tests.

Hi @xpavlic 
satosa-oidcop doens't have any dependency to pydantic, the exception you have shared, and thank you for that, doesn't expose any satosa-oidcop files

it seems to be a general satosa issue, can you confirm that?
Consider me available for any further check in satosa-oidcop

**Comment by xpavlic** [comment]
> > Following exception occurs, when I try to run tests.
> 
> Hi @xpavlic satosa-oidcop doens't have any dependency to pydantic, the exception you have shared, and thank you for that, doesn't expose any satosa-oidcop files
> 
> it seems to be a general satosa issue, can you confirm that? Consider me available for any further check in satosa-oidcop

Hi @peppelinux
The problem was, that I was installing dependencies with `python setup.py install` instead of `pip install .`  Since `setup.py install` is deprecated, everything should be ok.

I just want to mention, that satosa-oidcop has indirect dependency on pydantic.
Dependencies are flowing: satosa-oidcop -> SATOSA -> pyop -> oic -> pydantic.

**Comment by peppelinux** [comment]
Thank you guys, this sounds great

I look forward for any feedback before doing the release

**Comment by MdreW** [comment]
I have tested all on my production server and all work fine, great work @peppelinux :+1:

File: tests/test_oidcop.py [hunk 1]
```diff
@@ -214,27 +221,19 @@
             ],
             "userinfo_encryption_alg_values_supported": [
               "RSA-OAEP",
-              "RSA-OAEP-256",
-              "A192KW",
-              "A256KW",
-              "ECDH-ES",
-              "ECDH-ES+A128KW",
-              "ECDH-ES+A192KW",
```
Comment (MdreW) [inline tests/test_oidcop.py:222]: why do you want to remove elliptical encryption support? Is a good encryption method for me :)

**Review by MdreW — APPROVED** [review]
I have tested and all work fine, great work @peppelinux :+1: 
I'd rather not remove elliptical and aes encryption support from the configuration, but we can always add it later.

**Pull request merged**